packages/remirror__extension-text-color/src/text-color-extension.ts

```diff
@@ -115,6 +115,18 @@ export class TextColorExtension extends MarkExtension<TextColorOptions> {
     return this.store.commands.removeMark.original({ type: this.type, ...options, expand: true });
   }
 
+  /**
+   * Get the color for the selection
+   */
+  @helper()
+  getTextColorForSelection(selection?: Selection): Helper<string> {
+    if (!selection) return this.options.defaultColor;
+
+    const state = this.store.getState();
+    const textMarkInfo = getMarkRange(selection.$from, this.type);
+    if (!textMarkInfo) return  this.options.defaultColor;
```

Comment:
Formatting is a bit off here, suspect this would be fixed by running `pnpm fix`

```suggestion
    
    if (!textMarkInfo) {
      return this.options.defaultColor;
    }
    
```